binderhub/base.py

```diff
@@ -17,6 +19,23 @@ def initialize(self):
         if self.settings['auth_enabled']:
             self.hub_auth = HubOAuth.instance(config=self.settings['traitlets_config'])
 
+    def prepare(self):
+        super().prepare()
+        # check request ips early on all handlers
+        self.check_request_ip()
+
+    skip_check_request_ip = False
+
+    def check_request_ip(self):
+        """Check network block list, if any"""
+        if self.skip_check_request_ip or not self.settings.get("ban_networks"):
+            return
+        request_ip = ip_address(self.request.remote_ip)
+        for network in self.settings["ban_networks"]:
```

Comment:
If performance is too bad I found a nice trick in https://codereview.stackexchange.com/a/183221
Order the list of `ban_networks` by the integer representation of the first IP, then you can easily do a search to find the relevant one for `request_ip`.

Reply:
netaddr.IPSet [has an O(1) implementation](https://netaddr.readthedocs.io/en/latest/_modules/netaddr/ip/sets.html#IPSet.__contains__), assuming you can preconstruct a set (we can). I'm looking at comparing a version using the stdlib vs pulling in netaddr as a dependency.